src/core/uint.js

```diff
@@ -166,7 +186,7 @@ UInt.prototype.parse_generic = function(j) {
 };
 
 UInt.prototype.parse_hex = function(j) {
-  if (typeof j === 'string' && j.length === (this.constructor.width * 2)) {
+  if (new RegExp(`^[0-9A-F]{${this.constructor.width * 2}}$`).test(j)) {
```

Comment:
> I think UInt256.is_valid() is a good place to validate.

Not sure I agree, as UInt256 != InvoiceID, and InvoiceID is not a VL variable length (string) field. It only takes hex, not arbitrary characters, and that hex defines a number. ripple-lib even pads the hex strings passed along to a full 64 nibble string. 

Hrmmm.